merge.go

```diff
@@ -0,0 +1,599 @@
+	id               int64
+	typ              []string
+	props            map[string]interface{}
+	e                *lpg.Edge
+	fromNode, toNode int64
+}
+
+type updateEdge struct {
+	id               int64
+	typ              []string
+	props            map[string]interface{}
+	e                *lpg.Edge
+	fromNode, toNode int64
+}
+
+func buildQueriesFromDeltas(deltas []delta, cfg Config) ([]string, []map[string]interface{}) {
+	res := make([]string, 0)
+	vars := make([]map[string]interface{}, 0)
+	for _, d := range deltas {
+		switch d.operation {
+		case createNodeOp:
+			op := createNode{
+				n:      d.node,
+				id:     d.id,
+				labels: d.addLabel,
+				props:  d.setProp,
+			}
+			res = append(res, op.writeQuery(cfg))
+		case updateNodeOp:
+			op := updateNode{
+				n:      d.node,
+				id:     d.id,
+				labels: d.addLabel,
+				props:  d.setProp,
+			}
+			q, v := op.writeQuery(cfg)
+			res = append(res, q)
+			vars = append(vars, v)
+		case updateEdgeOp:
+			op := updateEdge{
+				e:     d.edge,
+				id:    d.id,
+				typ:   d.addLabel,
+				props: d.setProp,
+			}
+			res = append(res, op.writeQuery(cfg))
+		case createEdgeOp:
+			op := createEdge{
+				e:     d.edge,
+				id:    d.id,
+				typ:   d.addLabel,
+				props: d.setProp,
+			}
+			res = append(res, op.writeQuery(cfg))
+		}
+	}
+	return res, vars
+}
+
+type OperationQueue struct {
+	Ops  []string
+	vars []map[string]interface{}
+}
+
+func RunOperations(ctx *ls.Context, session *Session, tx neo4j.Transaction, ops OperationQueue) error {
+	for ix, op := range ops.Ops {
+		_, err := tx.Run(op, ops.vars[ix])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Merge(memGraph, dbGraph *lpg.Graph, dbGraphIds map[*lpg.Node]int64, dbEdges map[*lpg.Edge]int64, config Config) (*lpg.Graph, OperationQueue, error) {
+	memEntitiesMap := ls.GetEntityInfo(memGraph)
+	dbEntitiesMap := ls.GetEntityInfo(dbGraph)
+	findDBCounterpart := func(n *lpg.Node, schemaPV, idPV *ls.PropertyValue) *lpg.Node {
+		if ls.AsPropertyValue(n.GetProperty(ls.EntitySchemaTerm)).IsEqual(schemaPV) {
+			if ls.AsPropertyValue(n.GetProperty(ls.EntityIDTerm)).IsEqual(idPV) {
+				return n
+			}
+		}
+		return nil
+	}
+	memEntities := make([]*lpg.Node, len(memEntitiesMap))
+	dbEntities := make([]*lpg.Node, len(memEntitiesMap))
+	// align by index matching entity roots i.e, first entity in mem slice should match first entity in db slice
+	ix := 0
+	for n := range memEntitiesMap {
+		schemaPV := ls.AsPropertyValue(n.GetProperty(ls.EntitySchemaTerm))
+		eidPV := ls.AsPropertyValue(n.GetProperty(ls.EntityIDTerm))
+		memEntities[ix] = n
+		for db := range dbEntitiesMap {
+			if findDBCounterpart(db, schemaPV, eidPV) != nil {
+				dbEntities[ix] = db
+				delete(dbEntitiesMap, db)
+				continue
+			}
+		}
+		ix++
+	}
+	deltas := make([]delta, 0)
+	for ix := 0; ix < len(memEntities); ix++ {
+		d, _ := mergeEntity(memEntities[ix], dbEntities[ix], dbGraph, dbGraphIds, dbEdges)
+		deltas = append(deltas, d...)
+	}
```

Comment:
After the for-loop, find all newly created db entity roots (use nodeAssoc for that)
Find the nodes in the memgraph that is connected to the newly created  db node
Create the edges between them.